[Feature] Add HierText converter (#948)
open-mmlab/mmocr: docs/en/datasets/recog.md

## OpenVINO

> - Step1 (optional): Install [AWS CLI](#install-aws-cli-optional).

Reviewer: ```suggestion
- Step1 (optional): Install [AWS CLI](https://mmocr.readthedocs.io/en/latest/datasets/recog.html#install-aws-cli-optional).
```

  │   ├── train_label.jsonl
  │   └── val_label.jsonl (optional)
  ```

## HierText

- Step1 (optional): Install [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html). Here is a brief installation procedure, if you encounter problems, please check the official website.
  ```bash
  curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
  unzip awscliv2.zip
  sudo ./aws/install
  ./aws/install -i /usr/local/aws-cli -b /usr/local/bin
  !aws configure
  # this command will require you to input keys, you can skip them except
  # for the Default region name
  # AWS Access Key ID [None]:
  # AWS Secret Access Key [None]:
  # Default region name [None]: us-east-1

>   # Default output format [None]

Reviewer 2: Since there are also other datasets that use AWS CLI, we may create a new section after the Overview section (or somewhere), and link all these steps to that section.

![image](https://user-images.githubusercontent.com/45810070/165885357-ba4ef261-e534-451e-b170-2c98e284de2d.png)

And do not forget to update the link in other dataset that uses the AWS CLI

- Step2: Clone [HierText](https://github.com/google-research-datasets/hiertext) repo to get annotations
  ```bash
  mkdir HierText
  git clone https://github.com/google-research-datasets/hiertext.git
  ```
- Step3: Download `train.tgz`, `validation.tgz` from aws
  ```bash
  aws s3 --no-sign-request cp s3://open-images-dataset/ocr/train.tgz .
  aws s3 --no-sign-request cp s3://open-images-dataset/ocr/validation.tgz .
  ```
- Step4: Process raw data
  ```bash
  # process annotations
  mv hiertext/gt ./
  rm -rf hiertext
  mv gt annotations
  gzip -d annotations/train.jsonl.gz
  gzip -d annotations/validation.jsonl.gz
  # process images
  mkdir imgs
  mv train.tgz imgs/
  mv validation.tgz imgs/
  tar -xzvf imgs/train.tgz
  tar -xzvf imgs/validation.tgz
  ```

> - Step5: Generate `train_label.jsonl` and `val_label.jsonl`. HierText includes different levels of annotation, from paragraph, line, to word. Check the original [paper](https://arxiv.org/pdf/2203.15143.pdf) for details. E.g. set `--level paragraph` to get paragraph level's annotation. Set `--level line` to get line level annotation. set `--level word` to get word level's annotation.

Reviewer 2: ```suggestion
- Step5: Generate `train_label.jsonl` and `val_label.jsonl`. HierText includes different levels of annotation, including `paragraph`, `line`, and `word`. Check the original [paper](https://arxiv.org/pdf/2203.15143.pdf) for details. E.g. set `--level paragraph` to get paragraph-level annotation. Set `--level line` to get line-level annotation. set `--level word` to get word-level annotation.
```

  │   ├── train_label.jsonl
  │   └── val_label.jsonl (optional)
  ```

## HierText

> - Step1 (optional): Install [AWS CLI](#install-aws-cli-optional).

Reviewer: ```suggestion
- Step1 (optional): Install [AWS CLI](https://mmocr.readthedocs.io/en/latest/datasets/recog.html#install-aws-cli-optional).
```
